dotnet/src/MicroFocus.CafApi.QueryBuilder.Vqll.Parsers.SystemJson/SystemJsonFilterReader.cs

```diff
@@ -0,0 +1,541 @@
+/**
+ * Copyright 2022 Micro Focus or one of its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+using System;
+using System.Text.Json;
+using System.Text.Json.Nodes;
+using System.Collections.Generic;
+using Microsoft.Extensions.Logging;
+using System.Linq;
+
+namespace MicroFocus.CafApi.QueryBuilder.Vqll.Parsers.SystemJson
+{
+    public sealed class SystemJsonFilterReader
+    {
+        private readonly ILogger _logger;
+
+        private SystemJsonFilterReader(ILogger logger)
+        {
+            _logger = logger;
+        }
+
+        public static Filter<string> ReadFromJsonArray(JsonNode vqll, ILogger logger)
+        {
+            return new SystemJsonFilterReader(logger).CreateFilter(vqll);
+        }
+
+        private Filter<string> CreateFilter(JsonNode node)
+        {
+            _logger.LogDebug("Parsing: {0}", node);
```

Comment:
If we're going to log this then we should log in the Json form.

```suggestion
            if (_logger.IsEnabled(LogLevel.Debug))
            {
                _logger.LogDebug("Parsing: {0}", node.ToJsonString());
            }
```